### PR TITLE
build(client/electron): build Windows with MinGW

### PR DESCRIPTION
**Pull request by fortuna**
This works on both Linux and Windows.

I was even able to run tun2socks with Wine!

/cc @amircybersec

### CODE REVIEW

**Review by jyyi1 — CHANGES_REQUESTED** [review]
We need to update the CI script as well. The build jobs are failing.

Also suggest to update the PR title to `build(client/electron): ...`, [here is a list of all conventional commit types], we can either to `build` or `ci`.

**Review by fortuna — COMMENTED** [review]
Comments addressed

File: Makefile [hunk 1]
```diff
@@ -12,18 +12,17 @@ IMPORT_PATH=$(IMPORT_HOST)/Jigsaw-Code/outline-apps
...
 ROOT_PKG=client/src/tun2socks
-# Don't strip Android debug symbols so we can upload them to crash reporting tools.
-ANDROID_BUILD_CMD=$(GOBIND) -a -ldflags '-w' -target=android -androidapi 19 -tags android -work
 
 android: $(BUILDDIR)/android/tun2socks.aar
 
 $(BUILDDIR)/android/tun2socks.aar: $(GOMOBILE)
 	mkdir -p "$(BUILDDIR)/android"
-	$(ANDROID_BUILD_CMD) -o "$@" $(IMPORT_PATH)/$(ROOT_PKG)/outline/tun2socks $(IMPORT_PATH)/$(ROOT_PKG)/outline/shadowsocks
+  # Don't strip Android debug symbols so we can upload them to crash reporting tools.
```
Comment (jyyi1) [inline Makefile:20]: I think we are still stripping the debug symbols (`-w`)? Also I think we need to consider whether we should include debug symbols, that increase the binary size in a significant amount.

File: Makefile [hunk 2]
```diff
@@ -56,16 +55,13 @@ WINDOWS_BUILDDIR=$(BUILDDIR)/windows
...
 
-$(WINDOWS_BUILDDIR)/tun2socks.exe: $(XGO)
-	mkdir -p "$(WINDOWS_BUILDDIR)/$(IMPORT_PATH)"
-	$(XGO) -ldflags $(XGO_LDFLAGS) --targets=windows/386 -dest "$(WINDOWS_BUILDDIR)" -pkg $(ELECTRON_PKG) .
-	mv "$(WINDOWS_BUILDDIR)/$(IMPORT_PATH)-windows-386.exe" "$@"
-	rm -r "$(WINDOWS_BUILDDIR)/$(IMPORT_HOST)"
-
+$(WINDOWS_BUILDDIR)/tun2socks.exe:
+	mkdir -p "$(@D)"
+	GOOS=windows GOARCH=amd64 CGO_ENABLED=1 CC="x86_64-w64-mingw32-gcc" go build --ldflags=--extldflags=$(XGO_LDFLAGS) -o "$@" $(ELECTRON_PKG)
```
Comment (jyyi1) [inline Makefile:60]: Note that this is updating from a 32-bit (`386`) binary to 64-bit (`amd64`), which might break some old OS users (like 32-bit Windows 7 / 8 / 10).

Comment (jyyi1) [inline Makefile:60]: Maybe not in this PR: our CI is not checking the build file (it is building directly from native Windows), we may need to add some x-compile CI jobs in the future.

File: src/electron/README.md [hunk 3]
```diff
@@ -30,6 +29,26 @@ npm run action electron/start [windows|linux]
 
 ### Windows
 
+To build for Windows on a macOS, you need to install minGW v11.0.1.
```
Comment (jyyi1) [inline src/electron/README.md:32]:
```suggestion
To build for Windows on a macOS, you need to install [MinGW-w64](https://www.mingw-w64.org/) v11.0.1.
```

File: src/electron/README.md [hunk 4]
```diff
@@ -30,6 +29,26 @@ npm run action electron/start [windows|linux]
...
+
+With Homebrew (how to ensure consistent version?):
```
Comment (jyyi1) [inline src/electron/README.md:34]: Will MacPorts allow you downloading an old version? Actually homebrew is not the official release channel, MacPorts is: https://www.mingw-w64.org/downloads/

**Review by jyyi1 — COMMENTED** [review]
This is exiciting! We can now finally cross-compile Windows binaries **natively**! Before merging the code, we need to consider some behavior changes carefully (for example, changing from 32-bit to 64-bit).

File: Makefile (hunk 1 above)
Comment (fortuna) [reply Makefile:20]: -s is what disables the symbol table.
-w is disable DWARF generation

Ideally we would output tun2socks with debug info, and the client build would remove them. Is that possible?

But that's out of scope for this PR

File: Makefile (hunk 2 above)
Comment (fortuna) [reply Makefile:60]: As discussed, let's move forward with 64-bit

File: src/electron/README.md (hunk 3 above)
Comment (fortuna) [reply src/electron/README.md]: Done

File: src/electron/README.md (hunk 4 above)
Comment (fortuna) [reply src/electron/README.md]: Yes, updated.

**Review by fortuna — COMMENTED** [review]
Another look?

**Review by jyyi1 — APPROVED** [review]
LGTM now, thanks!

**Pull request merged**